src/controllers/userController.ts

```diff
@@ -173,3 +175,118 @@ export const updateUserProfile = async (req: Request, res: Response) => {
         })
     }
 }
+
+
+export const resetPasswordEmail = async (req: Request, res: Response) => {
+    const { email } = req.body;
+
+    try {
+        const user = await findUserByEmail(email);
+
+        if (!user) {
+            return res.status(404).json({
+                success: false,
+                message: 'User not found',
+            });
+        }
+
+        const randomToken = Math.random().toString(36).substring(2, 22);
+        const resetToken = sha256(randomToken);
+        const resetTokenExpiration = Date.now() + 3600000;
+        const newToken = await UserToken.create({
+            userId: user.id,
```

Comment:
I would appreciate it if this code would be moved to another file, or to another function to separate the concerns and since this function is long

Reply:
done